addons/website/static/src/js/utils.js

```diff
@@ -23,12 +23,26 @@ function autocompleteWithPages(self, $input) {
                     limit: 15,
                 },
             }).then(function (exists) {
-                var rs = _.map(exists, function (r) {
-                    return r.loc;
-                });
-                response(rs);
+                if ($input.val()[0] === '#') {
+                    var anchors = [];
+                    var def = $.when(document.body.outerHTML);
+                    def.then(function (response) {
+                        _.each($(response).find('[id][data-anchor="true"]'), function (el) {
+                            anchors.push('#' + el.id);
+                        });
```

Comment:
I block a bit on this one, if you have one more hint for me when you have time @qsm-odoo

Reply:
This code is a duplicate of what you implemented directly for the anchor input. What you should do is share that code. What I have in mind is an `options` parameter to this `autocompleteWithPages` function so that it proposes pages or anchor or both. Then the URL input in the link dialog will use both functionalities and the anchor input will use anchor only (by calling this `autocompleteWithPages` function with the right parameters when initializing the `<input/>`s) . That way your code is reusable and more importantly not duplicated.